packages/core/strapi/lib/services/entity-validator/validators.js

```diff
@@ -71,13 +71,33 @@ const addMaxFloatValidator = ({ max }, validator) =>
 const addStringRegexValidator = ({ regex }, validator) =>
   _.isUndefined(regex) ? validator : validator.matches(new RegExp(regex));
 
+const addUniqueValidator = (attr, validator, { uid, attributeName, entityId }) => {
+  if (attr.unique) {
+    return validator.test('unique', 'This attribute must be unique', async value => {
+      let whereParams = entityId
+        ? { $and: [{ [attributeName]: value }, { $not: { id: entityId } }] }
+        : { [attributeName]: value };
+
+      const record = await strapi.db.query(uid).findOne({
+        select: ['id', attributeName],
```

Comment:
do you need attributeName ?

Reply:
Forgot to remove that. It's a remnant of a test 🤦‍♂️